### PR TITLE
Updated all deps except k8s.io/client.

### PR DESCRIPTION
**Pull request by bwplotka**
[Circle CI does not allow us to update k8s.io/client-go for some reasons,](https://github.com/prometheus/prometheus/pull/7158) trying without that dep, updated rest to latest/

(step 0 for release: of https://github.com/prometheus/prometheus/blob/master/RELEASE.md#0-updating-dependencies)

Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
FYI: swagger output will be different now:

```
diff --git a/documentation/dev/api/swagger.json b/documentation/dev/api/swagger.json
index 29fddaf9..efba1e7f 100644
--- a/documentation/dev/api/swagger.json
+++ b/documentation/dev/api/swagger.json
@@ -14,7 +14,7 @@
     "/v2/admin/tsdb/clean_tombstones": {
       "post": {
         "summary": "CleanTombstones removes the deleted data from disk and cleans up the existing tombstones.",
-        "operationId": "TSDBCleanTombstones",
+        "operationId": "Admin_TSDBCleanTombstones",
         "responses": {
           "200": {
             "description": "A successful response.",
@@ -37,7 +37,7 @@
     "/v2/admin/tsdb/delete_series": {
       "post": {
         "summary": "DeleteSeries deletes data for a selection of series in a time range.",
-        "operationId": "DeleteSeries",
+        "operationId": "Admin_DeleteSeries",
         "responses": {
           "200": {
             "description": "A successful response.",
@@ -70,7 +70,7 @@
     "/v2/admin/tsdb/snapshot": {
       "post": {
         "summary": "Snapshot creates a snapshot of all current data into 'snapshots/\u003cdatetime\u003e-\u003crand\u003e' under the TSDB's data directory.",
-        "operationId": "TSDBSnapshot",
+        "operationId": "Admin_TSDBSnapshot",
         "responses": {
           "200": {
             "description": "A successful response.",

Exited with code exit status 1
``` 

Is this breaking change?

**Comment by brian-brazil** [comment]
It's not, that entire interface is marked as experimental and is slated for deletion.

**Comment by brian-brazil** [comment]
Why is pool_test.go changing? I don't see anything dependency related there.

**Comment by bwplotka** [comment]
@brian-brazil Flaky test that was just introduced to master

**Comment by bwplotka** [comment]
@juliusv  @brian-brazil @roidelapluie can we merge it and cut release? :hugs:

File: go.mod [hunk 1]
```diff
@@ -38,29 +55,40 @@ require (
...
+	k8s.io/client-go v0.17.5
 	k8s.io/klog v1.0.0
+	k8s.io/utils v0.0.0-20200414100711-2df71ebbae66 // indirect
 )
 
-replace k8s.io/klog => github.com/simonpasquier/klog-gokit v0.1.0
+replace (
+	// TODO(bwplotka): Remove once client-go will move to renamed openapiv2 package
+	// (https://github.com/kubernetes/test-infra/issues/16116).
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.0
```
Comment (roidelapluie) [inline go.mod:92]: Then just keep v0.4.0 above

Comment (bwplotka) [reply go.mod:92]: noooo and 2h lost again ;p

Comment (bwplotka) [reply go.mod:92]: Actually that is correct:

```
github.com/prometheus/prometheus/discovery/kubernetes imports
	k8s.io/client-go/kubernetes imports
	k8s.io/client-go/discovery imports
	github.com/googleapis/gnostic/OpenAPIv2: module github.com/googleapis/gnostic@latest (v0.4.1) found, but does not contain package github.com/googleapis/gnostic/OpenAPIv2
```

Without this.

Comment (roidelapluie) [reply go.mod:92]: with v0.4.0 ? I don't think so, I did that the last time

Comment (bwplotka) [reply go.mod:92]: Wait what do you mean exactly? I thought you suggested to remove this explicit replace.. or what?

File: go.mod [hunk 2]
```diff
@@ -3,30 +3,47 @@ module github.com/prometheus/prometheus
...
+	github.com/go-openapi/swag v0.19.9 // indirect
+	github.com/go-openapi/validate v0.19.8 // indirect
 	github.com/gogo/protobuf v1.3.1
+	github.com/golang/protobuf v1.4.0 // indirect
 	github.com/golang/snappy v0.0.1
-	github.com/google/pprof v0.0.0-20200229191704-1ebb73c60ed3
-	github.com/gophercloud/gophercloud v0.8.0
-	github.com/grpc-ecosystem/grpc-gateway v1.14.1
+	github.com/google/pprof v0.0.0-20200417002340-c6e0a841f49a
+	github.com/googleapis/gnostic v0.4.1 // indirect
```
Comment (roidelapluie) [inline go.mod:32]:
```suggestion
	github.com/googleapis/gnostic v0.4.0 // indirect
```

Comment (roidelapluie) [reply go.mod:32]: this :)

Comment (bwplotka) [reply go.mod:32]: :+1:

Comment (bwplotka) [reply go.mod]: That worked, thanks @roidelapluie

**Comment by bwplotka** [comment]
Should be good to go, when green :hugs:  @roidelapluie

**Comment by simonpasquier** [comment]
My 2 cents: `go get -u ./...` is bit too coarse as it updates both direct & indirect dependencies. You can run `sed -i /indirect/d go.mod && go mod tidy && go mod vendor` to trim down the dep upgrades. I'll update the release documentation with the correct instructions (I forgot to do this after v2.17.0).

**Comment by brian-brazil** [comment]
:+1: 

We really shouldn't be making unrelated test fixes in a dep update PR, please ensure that that is a separate commit from the rest which can be squashed.

**Comment by bwplotka** [comment]
Agree, just being lazy here ):

Doing normal merge without squash now. Thanks @brian-brazil

**Comment by zetaab** [comment]
@bwplotka may I ask when prometheus is going to update libraries back to use 0.18.2 deps? The problem now is that we cannot use anymore this repository as library dependency in another application


```
# github.com/prometheus/prometheus/discovery/kubernetes
vendor/github.com/prometheus/prometheus/discovery/kubernetes/client_metrics.go:113:18: too many arguments in call to metrics.Register
	have (*clientGoRequestMetricAdapter, *clientGoRequestMetricAdapter)
	want (metrics.RegisterOpts)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:320:19: not enough arguments in call to e.List
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:325:20: not enough arguments in call to e.Watch
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:333:19: not enough arguments in call to s.List
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:338:20: not enough arguments in call to s.Watch
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:346:19: not enough arguments in call to p.List
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:351:20: not enough arguments in call to p.Watch
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:372:19: not enough arguments in call to p.List
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:377:20: not enough arguments in call to p.Watch
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:394:19: not enough arguments in call to s.List
	have ("k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
	want (context.Context, "k8s.io/apimachinery/pkg/apis/meta/v1".ListOptions)
vendor/github.com/prometheus/prometheus/discovery/kubernetes/kubernetes.go:394:19: too many errors
```

**Pull request merged**